drivers/rilmodem/radio-settings.c

```diff
@@ -286,20 +687,90 @@ void ril_query_available_rats(struct ofono_radio_settings *rs,
 	}
 }
 
-void ril_delayed_register(const struct ofono_error *error, void *user_data)
+static void gprs_watch_cb(struct ofono_atom *atom,
+				enum ofono_atom_watch_condition cond,
+				void *data)
 {
-	struct ofono_radio_settings *rs = user_data;
+	struct ofono_radio_settings *rs = data;
+
+	if (cond != OFONO_ATOM_WATCH_CONDITION_REGISTERED)
+		return;
+
+	set_ia_apn(rs);
+}
+
+static void set_safe_preferred_cb(const struct ofono_error *error, void *data)
+{
+	if (error->type != OFONO_ERROR_TYPE_NO_ERROR) {
+		ofono_error("%s: error setting radio access mode", __func__);
+
+		return;
+	}
+}
+
+static void radio_settings_register(const struct ofono_error *error,
+					unsigned int available_rats,
+					void *data)
+{
+	struct ofono_radio_settings *rs = data;
+	struct radio_data *rd = ofono_radio_settings_get_data(rs);
+
+	g_ril_register(rd->ril, RIL_UNSOL_RADIO_CAPABILITY,
+							radio_caps_event, rd);
 
 	if (error->type != OFONO_ERROR_TYPE_NO_ERROR)
 		ofono_error("%s: cannot set default fast dormancy", __func__);
```

Comment:
This looks like leftover code from the old ril_delayed_register, and should be removed as it doesn't make any sense in this context.